spec.html

```diff
@@ -82,8 +71,15 @@ <h1>Array.prototype.reversed ( )</h1>
 
                 <emu-alg>
                     1. Let _O_ be ? ToObject(*this* value).
-                    1. Let _A_ be ArrayClone(_O_).
-                    1. Perform ? Call(Array.prototype.reverse, _A_, «»).
+                    1. Let _len_ be ? LengthOfArrayLike(_O_).
+                    1. Let _A_ be ? ArraySpeciesCreate(_O_, 𝔽(_len_)).
+                    1. Let _k_ be 0.
+                    1. Repeat, while _k_ &lt; _len_,
+                        1. Let _from_ be ! ToString(𝔽(_len_ - _k_ - 1)).

```

Comment:
and perhaps used here as well with a "direction" argument?

Reply:
I like it. Are you imagining a direction like 'forwards'/'backwards' or a 'step amount' like +1/-1 ?

Reply:
Probably a step amount? Whatever seems cleaner.